pyop2/mpi.py

```diff
@@ -465,26 +474,41 @@ def compilation_comm(comm):
     else:
         comp_comm = comm
     incref(comp_comm)
+    weakref.finalize(obj, decref, comp_comm)
     return comp_comm
 
 
+def finalize_safe_debug():
+    if PYOP2_FINALIZED:
+        if logger.level > DEBUG and not _running_on_ci:
+            debug = lambda string: None
+        else:
+            debug = lambda string: print(string)
+    return debug
```

Comment:
I assume this is a global variable? Otherwise this will break.

Reply:
Yes, it's global